server/src/main/java/org/elasticsearch/script/CtxMap.java

```diff
@@ -52,20 +50,15 @@ protected CtxMap(Map<String, Object> source, Metadata metadata) {
         }
     }
 
-    protected Map<String, Object> wrapSource(Map<String, Object> source) {
-        Map<String, Object> wrapper = Maps.newHashMapWithExpectedSize(1);
-        wrapper.put(SOURCE, source);
-        return wrapper;
+    protected boolean directSourceAccess() {
```

Comment:
Can you please add a short javadoc describing what this controls (ie mentioning whether source is indirectly accessed through the "_source" key or at the root of ctx).

Reply:
Done.